.jenkins/cscs/env-clang-16.sh

```diff
@@ -4,7 +4,7 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
-cxx_std="20"
+cxx_std="23"
```

Comment:
Did you test this with the spack package? Doesn't the spack package require updating first? I'm happy to make the change for the Piz Daint spack instance until it's changed upstream. It looks like the boost and fmt packages also need a 23 `cxxstd` added. They'd need to be tested as well.